python/publish/__init__.py

```diff
@@ -832,20 +833,21 @@ def get_case_annotation(messages: CaseMessages,
 
 
 def get_case_annotations(case_results: UnitTestCaseResults,
-                         report_individual_runs: bool) -> List[Annotation]:
+                         report_individual_runs: bool,
+                         test_file_path_prefix: str) -> List[Annotation]:
     messages = get_case_messages(case_results)
     return [
-        get_case_annotation(messages, key, state, message, report_individual_runs)
+        get_case_annotation(messages, key, state, message, report_individual_runs, test_file_path_prefix)
         for key in messages
         for state in messages[key] if state not in ['success', 'skipped']
         for message in (messages[key][state] if report_individual_runs else
                         [list(messages[key][state].keys())[0]])
     ]
 
 
-def get_error_annotation(error: ParseError) -> Annotation:
+def get_error_annotation(error: ParseError, test_path_file_prefix: str) -> Annotation:
     return Annotation(
-        path=error.file,
+        path=error.file.replace(test_path_file_prefix, '', 1),
```

Comment:
This would also remove the prefix from the test file if it does not occur as a prefix, Not very likely but better be safe:
```suggestion
        path=error.file[len(test_path_file_prefix):] if test_path_file_prefix and error.file.startswith(test_path_file_prefix) else error.file
```